Add pull provider documentation (#1548)
theforeman/foreman-documentation: guides/common/modules/proc_setting-up-pull-client.adoc

[id="setting-up-pull-client_{context}"]
= Setting Up the Remote Execution Pull Client

For {SmartProxies} in `pull-mqtt` mode, hosts can subscribe to remote jobs using the remote execution pull client.
No SSH connection is required.

Use this procedure to configure a host to use the pull client:

.Prerequisites
* The host must be registered to {ProjectName}.
* The host's {SmartProxy} must be in `pull-mqtt` mode.
* The {Team} {project-client-name} repository must be enabled and synchronized on {ProjectServer}, and enabled on the host.

.Procedure
. On the host, install the `katello-pull-transport-migrate` RPM package:
+
----
# yum install katello-pull-transport-migrate
----

> The host's `subscription-manager` configuration is used to configure the `yggdrasil` client on the host, and the worker is started.

Author: @adamruzicka @ezr-ondrej

> The host's `subscription-manager` configuration is used to configure the `yggdrasil` client on the host, and the worker is started.

Author: @ehelms ?

> The host's `subscription-manager` configuration is used to configure the `yggdrasil` client on the host, and the worker is started.

Reviewer: Afaik it will still be `yggdrasil`

[id="setting-up-pull-client_{context}"]
= Setting Up the Remote Execution Pull Client

For {SmartProxies} in `pull-mqtt` mode, hosts can subscribe to remote jobs using the remote execution pull client.
No SSH connection is required.

Use this procedure to configure a host to use the pull client:

.Prerequisites
* The host is registered to {ProjectName}.
* The host's {SmartProxy} is in `pull-mqtt` mode.
* The {Team} {project-client-name} repository is enabled and synchronized on {ProjectServer}, and enabled on the host.

> * The host is able to communicate with its {SmartProxy} over MQTT using port 1883.

Reviewer 2: While it should be obvious, we also should make note that the host must be able to communicate with the {SmartProxy} over HTTP.

> [id="setting-up-pull-client_{context}"]

Reviewer 3: title does not match anchor and file name. Please adjust this.

Author: sorry, I'm not very good at asciidoc yet. Could you please clarify what I need to change to fix this?

Reviewer 3: Absolutely, thanks for asking! In general, we ensure that the file name, the anchor, and the title "match" (minus attributes and the type prefix such as `con_` or `proc_`).

For your example:

Anchor: `[id="Setting_Up_the_Remote_Execution_Pull_Client_{context}"]` (without prefix, with `{context}` suffix)
Title: `Setting Up the Remote Execution Pull Client` (original)
File name: `proc_setting-up-the-remote-execution-pull-client.adoc` (all lowercase; only single underscore)

If you want to adjust the title, then I'd kindly ask you to also adjust the anchor and file name.

Author: Changed file name and anchor so they match the new title.

= Setting Up the Remote Execution Pull Client

For {SmartProxies} in `pull-mqtt` mode, hosts can subscribe to remote jobs using the remote execution pull client.

> No SSH connection is required.

Reviewer 3: ```suggestion
Managed hosts do not require an SSH connection to their {SmartProxyServer}.
```

> Use this procedure to configure a host to use the pull client:

Reviewer 3: This line does not really match the title. Maybe reword the title and drop this line.

.Prerequisites

> * The host is registered to {ProjectName}.

Reviewer 3: Maybe reword to "You have registered the host to {Project}."?

> * The host's {SmartProxy} is in `pull-mqtt` mode.

Reviewer 3: Please rephrase.

* The {Team} {project-client-name} repository is enabled and synchronized on {ProjectServer}, and enabled on the host.

> * The host is able to communicate with its {SmartProxy} over MQTT using port 1883.

Reviewer 3: ```suggestion
* The host is able to communicate with its {SmartProxy} over MQTT using port `1883`.
```

* The host is able to communicate with its {SmartProxy} over HTTPS.

.Procedure

> . On the host, install the `katello-pull-transport-migrate` RPM package:

Reviewer 3: ```suggestion
. Install the `katello-pull-transport-migrate` package on your managed host:
```

+
----

> # yum install katello-pull-transport-migrate

Reviewer 3: This seems to depend on the OS/EL version. what about el8? debian? Is this EL only?

Author: `subscription-manager` is required, so I'd say yes it's EL-only.

Reviewer 3: FYI: You can also run subscription-manager on sles and Debian/Ubuntu: https://apt.atix.de/

So maybe make the "yum" conditional?

Author: @adamruzicka @ehelms Is the pull client supported on Debian?

Reviewer: I'd be inclined to say no. The binaries we build are static and should run essentially anywhere as long as the platform and architecture match, but afaik we don't build packages for debian.

Reviewer 3: Thanks for the info. I guess the suggestion remains to split the "yum install" command into EL7 and EL8/9 (dnf).

----

> The host's `subscription-manager` configuration and consumer certificates are used to configure the `yggdrasil` client on the host, and the worker is started.

Reviewer 4: "the worker" -> "the yggdrasild worker" or "pull mode client worker"? (we speak about 'worker' for the first time)
